src/sentry/issues/occurrence_consumer.py

```diff
@@ -356,3 +361,50 @@ def _process_message(
             txn.set_tag("result", "error")
             raise InvalidEventPayloadError(e)
     return None
+
+
+def _process_batch(worker: ThreadPoolExecutor, message: Message[ValuesBatch[KafkaPayload]]):
+    """
+    Receives batches of occurrences. This function will take the batch
+    and group them together by fingerprint (ensuring order is preserved) and
+    execute each group using a ThreadPoolWorker.
+
+    By batching we're able to process occurrences in parallel while guaranteeing
+    that no occurrences are processed out of order per group.
+    """
+    batch = message.payload
+
+    occcurrence_mapping: Mapping[str, list[Mapping[str, Any]]] = defaultdict(list)
+
+    for item in batch:
+        assert isinstance(item, BrokerValue)
+
+        try:
+            payload = json.loads(item.payload.value, use_rapid_json=True)
+        except Exception:
+            logger.exception("Failed to unpack message payload")
+            continue
+        # group by the fingerprint, there should only be one of them
+        partition_key: str = payload["fingerprint"][0] if payload["fingerprint"] else ""
+        occcurrence_mapping[partition_key].append(payload)
+
+    # Number of occurrences that are being processed in this batch
+    metrics.gauge("occurrence_consumer.checkin.parallel_batch_count", len(batch))
+
+    # Number of groups we've collected to be processed in parallel
+    metrics.gauge("occurrence_consumer.checkin.parallel_batch_groups", len(occcurrence_mapping))
+    # Submit occurrences for processing
+    with sentry_sdk.start_transaction(op="process_batch", name="occurrence.occurrence_consumer"):
```

Comment:
i think with how our sdk instruments threads, this won't work very well but can see data in prod and go from there